pkg/controller/gardenletdeployer/actuator.go

```diff
@@ -700,6 +701,12 @@ func PrepareGardenletChartValues(
 		)
 	}
 
+	// gardenlet should manage Lease object in the namespace it is deployed to
+	if gardenletConfig.LeaderElection == nil {
+		gardenletConfig.LeaderElection = &componentbaseconfigv1alpha1.LeaderElectionConfiguration{}
+	}
+	gardenletConfig.LeaderElection.ResourceNamespace = gardenNamespaceTargetCluster
```

Comment:
What is `gardenNamespaceTargetCluster`? I assume that it is the `garden` namespace in an ordinary seed scenario inside the seed cluster and `kube-system` in the autonomous shoot cluster. If so, would it make sense to rename the variable to `gardenletNamespaceTargetCluster`?